### PR TITLE
Start of a more comprehensive duplicate test

### PR DESCRIPTION
**Pull request by cschroedl-gov**
Something is still wrong with it, but it may just be in the error reporting. Feel free to hack on this if you get a chance before I do.

### CODE REVIEW

File: tests/testthat/test_may_load.R [hunk 1]
```diff
@@ -105,12 +105,18 @@ test_that("Load data have the correct number of significant digits", {
...
-                                 length(unique(paste(temp_mloads$SITE_ABB,temp_mloads$CONSTIT,temp_mloads$mod1,temp_mloads$WY,sep="_")))==nrow(temp_mloads)   
-  )
-  
-  expect_no_errors(result)
-  
+
+   mloads_without_ignored_modtypes <- subset(mloads, !(MODTYPE %in% c('COMP', 'CONTIN')))
+   unique_columns <- mloads_without_ignored_modtypes[c('SITE_QW_ID', 'CONSTIT', 'WY')]
+   duplicates <- unique_columns[duplicated(unique_columns),]
+   if(nrow(duplicates) > 0) {
```
Comment (cschroedl-gov) [inline tests/testthat/test_may_load.R:112]: The intent of this conditional block is to prepare a helpful error message - one that starts with a descriptive summary and ends with R's handy native representation of the data frame containing duplicates.

```
There were duplicate entries that shared the following values:
       SITE_QW_ID CONSTIT   WY
106864   07373420      TP 2014
```

File: tests/testthat/test_may_load.R [hunk 2]
```diff
@@ -105,12 +105,18 @@ test_that("Load data have the correct number of significant digits", {
...
-  result <- validate::check_that(temp_mloads, 
-                                
-                                 length(unique(paste(temp_mloads$SITE_ABB,temp_mloads$CONSTIT,temp_mloads$mod1,temp_mloads$WY,sep="_")))==nrow(temp_mloads)   
-  )
-  
-  expect_no_errors(result)
-  
+
+   mloads_without_ignored_modtypes <- subset(mloads, !(MODTYPE %in% c('COMP', 'CONTIN')))
+   unique_columns <- mloads_without_ignored_modtypes[c('SITE_QW_ID', 'CONSTIT', 'WY')]
```
Comment (cschroedl-gov) [inline tests/testthat/test_may_load.R:110]: I forget, is our unique key on this data frame ('SITE_QW_ID', 'CONSTIT', 'WY'), or do we need something more?

**Comment by carlschroedl** [comment]
I fixed the bug in the error reporting. Will push the fix when I get off the bus in ~ 30 min.

**Comment by cschroedl-gov** [comment]
Closing because changes were incorporated in https://github.com/USGS-CIDA/nar_data/pull/39

**Pull request closed without merging**